kinto/core/storage/postgresql/__init__.py

```diff
@@ -637,15 +636,15 @@ def get_all(self, collection_id, parent_id, filters=None, sorting=None,
             safeholders['pagination_rules'] = 'WHERE {}'.format(sql)
             placeholders.update(**holders)
 
-        if limit:
-            # We validate the limit value in the resource class as integer.
-            safeholders['pagination_limit'] = 'LIMIT {}'.format(limit)
+        limit = min(self._max_fetch_size, limit) if limit else self._max_fetch_size
+        # We validate the limit value in the resource class as integer.
+        safeholders['pagination_limit'] = 'LIMIT {}'.format(limit)
```

Comment:
We don't actually need this to be a `safeholder` any more, we can put `LIMIT` in the query.